src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,67 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = '', health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.findIndex((animal) => animal === this);
+
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden;
+
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
The hide method should only set the 'hidden' property to true according to the task description. If there's a requirement for a method to toggle the property, consider creating a separate method for that purpose.